ecosystem/nft-metadata-crawler/nft-metadata-crawler-utils/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright © Aptos Foundation
+
+use chrono::{NaiveDateTime, Utc};
+use google_cloud_auth::{
+    project::{create_token_source, Config},
+    token_source::TokenSource,
+};
+
+pub mod gcs;
+pub mod pubsub;
+
+/**
+ * Struct to help with parsing of CSV
+ */
+#[derive(Clone, Debug)]
+pub struct NFTMetadataCrawlerEntry {
+    pub token_data_id: String,
+    pub token_uri: String,
+    pub last_transaction_version: i32,
+    pub last_transaction_timestamp: chrono::NaiveDateTime,
+    pub last_updated: chrono::NaiveDateTime,
+    pub force: bool,
+}
+
+impl NFTMetadataCrawlerEntry {
+    pub fn new(s: String) -> anyhow::Result<Self> {
```

Comment:
Hmm it might be helpful to explain what's going on. What is S and why are we doing splits?